core/controlplane/config/templates/cloud-config-controller

```diff
@@ -292,9 +292,10 @@ coreos:
         {{ if .KubeDns.NodeLocalResolver }}--cluster-dns=${COREOS_PRIVATE_IPV4} \
         {{ else }}--cluster-dns={{.DNSServiceIP}} \
         {{ end }}--cluster-domain=cluster.local \
-        --cloud-provider=aws \{{if .Experimental.Admission.Priority.Enabled}}
+        --cloud-provider=aws \
+        {{if .Experimental.Admission.Priority.Enabled}}
         --feature-gates=PodPriority=true \
-        {{end}}\
+        {{end}}
```

Comment:
Doesn't this line also has to end with `\`, or otherwise anything passed into `$KUBELET_OPTS` will be considered to be part of the systemd configuration itself?

Reply:
@danielfm maybe when `feature-gates` it's enabled works correct, but when it's disabled, this is how it looks:

```bash
--register-with-taints=node.alpha.kubernetes.io/role=master:NoSchedule \
--allow-privileged=true \
--pod-manifest-path=/etc/kubernetes/manifests \
--cluster-dns=10.3.0.10 \
--cluster-domain=cluster.local \
--cloud-provider=aws \\
$KUBELET_OPTS
Restart=always
RestartSec=10
```

Reply:
The double backslash prevent `kubelet.service` to start

Reply:
Since you dropped the `{{ if }}` to the next line, wouldn't it look like this then (assuming you put the `\` after the `{{ end }}`?

```
        ...
        --cloud-provider=aws \
        \
        $KUBELET_OPTS
```

...which I believe is correct.

Anyway, we need to make sure this change doesn't break things for people relying on `$KUBELET_OPTS`.

Reply:
I'll test it again with and without `--feature-gates`

Reply:
> Since you dropped the {{ if }} to the next line, wouldn't it look like this then (assuming you put the \ after the {{ end }}?

Yes, by dropping the `{{ if }}` to the next line, it's also dropping the `\` to the next line.

Reply:
```yaml
    priority:
      enabled: false
```
```bash
....
        --pod-manifest-path=/etc/kubernetes/manifests \
        --cluster-dns=10.3.0.10 \
        --cluster-domain=cluster.local \
        --cloud-provider=aws \
        
        $KUBELET_OPTS
.....
```

```yaml
    priority:
      enabled: true
```
```bash
....
        --pod-manifest-path=/etc/kubernetes/manifests \
        --cluster-dns=10.3.0.10 \
        --cluster-domain=cluster.local \
        --cloud-provider=aws \
        
        --feature-gates=PodPriority=true \
        
        $KUBELET_OPTS
....
```

Reply:
I think that dropping the `{{if}}`  should be enough

Reply:
After the cade6e91b4893787e16fed55324a3527f6d80fc1 looks like this:
```yaml
    priority:
      enabled: false
```
```bash
....
        --pod-manifest-path=/etc/kubernetes/manifests \
        --cluster-dns=10.3.0.10 \
        --cluster-domain=cluster.local \
        --cloud-provider=aws \
        \
        $KUBELET_OPTS
.....
```

```yaml
    priority:
      enabled: true
```
```bash
....
        --pod-manifest-path=/etc/kubernetes/manifests \
        --cluster-dns=10.3.0.10 \
        --cluster-domain=cluster.local \
        --cloud-provider=aws \
        
        --feature-gates=PodPriority=true \
        \
        $KUBELET_OPTS
....
```

Reply:
Tested and `kubelet.service` starts fine now.